test/functional/normal/put_spec.lua

```diff
@@ -0,0 +1,1012 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local insert = helpers.insert
+local feed = helpers.feed
+local expect = helpers.expect
+local eq = helpers.eq
+local execute = helpers.execute
+local curbuf_contents = helpers.curbuf_contents
+local funcs = helpers.funcs
+local dedent = helpers.dedent
+local getreg = funcs.getreg
+
+local function reset()
+    clear()
+    insert([[
+    Line of words 1
+    Line of words 2]])
+    execute('goto 1')
+    feed('itest_string.<esc>u')
+    funcs.setreg('a', 'test_stringa', 'V')
+    funcs.setreg('b', 'test_stringb\ntest_stringb\ntest_stringb', 'b')
+    funcs.setreg('"', 'test_string"', 'v')
+end
+
+local function map(func, tab)
+    local rettab = {}
+    for k, v in pairs(tab) do
+        rettab[k] = func(v)
+    end
+    return rettab
+end
+
+local function filter(filter_func, tab)
+    local rettab = {}
+    for _, entry in pairs(tab) do
+        if not filter_func(entry) then
+            table.insert(rettab, entry)
+        end
+    end
+    return rettab
+end
+
+local function splitlines(string)
```

Comment:
Is this like the VimL `split()` function? If possible it would be better to use `helpers.funcs.split()` which returns a lua table. If we have ad-hoc functions like this in the tests they're difficult to follow, one must check the implementation instead of seeing immediately the familiar VimL function.